cpp/oneapi/dal/backend/linalg/matrix.hpp

```diff
@@ -48,6 +49,7 @@ class shape {
         }
         shape_[0] = row_count;
         shape_[1] = column_count;
+        ONEDAL_ASSERT(count() / columns() == rows(), "Shape count overflow");
```

Comment:
Shouldn't we do it by using `integer_overflow_ops<Data>::check_mul_overflow`?

Reply:
Assertion is enough in this case.